VueJS/index.js

```diff
@@ -3,11 +3,37 @@ import ListItem from './components/ListItem.js'
 
 new Vue({
   el: '#app',
+  components: {
+    List,
+    ListItem
+  },
   data: {
-    todos: [
+    lastId: 3,
+    todos:  [
       { id: '1', message: 'Foo', },
       { id: '2', message: 'Bar', },
       { id: '3', message: 'Baz', }
     ]
-  }
+  },
+  methods: {
+    newTodo: function() {      
+      this.lastId++
+      this.todos.push({id : this.lastId, message: ""})
+      // console.log("New Todo: [" + this.lastId + "]")     
+    },  
+    updateTodo: function(passedTodo) {
+        // console.log("(Großeltern) \nIch erhielt die todo [" + passedTodo.id + " | " + passedTodo.message + "] via $emit ")
+        // console.log("(Großeltern) \nFinde den entsprechenden index passend zur ID(" + passedTodo.id +") in todos[]")
+        let index = this.todos.findIndex( todo => todo.id == passedTodo.id)
+        // console.log("(Großeltern) \nUpdate das entsprechend gefundene Objekt in todos[]")
+        this.todos[index] = passedTodo   
```

Comment:
First time I see sb. replacing the whole object, not only the property of the object. Great! :clap: 

You can do a one-liner here:
```js
this.todos = this.todos.map(t => t.id === passedTodo.id ? passedTodo : t)
```